httomo/runner/loader.py

```diff
@@ -261,27 +267,18 @@ def _calculate_chunk_shape(
             self._global_shape[2],
         )
 
-    def _get_h5py_dataset(self) -> h5py.Dataset:
-        """
-        Get an h5py `Dataset` object that represents the data being loaded
-        """
-        f = h5py.File(self._in_file, "r", driver="mpio", comm=self._comm)
-        dataset: h5py.Dataset = f[self._data_path]
-        return dataset
-
     def read_block(self, start: int, length: int) -> DataSetBlock:
         block = self._data.make_block(self._slicing_dim, start, length)
         return block
 
     # NOTE: This method is largely copied from `load.get_data_indices()`; that function should
     # be removed in the future if/when `StandardTomoLoader` gets merged.
     def _get_data_indices(self) -> List[int]:
-        with h5py.File(self._in_file, "r", driver="mpio", comm=self._comm) as f:
-            if self._image_key_path is None:
-                data: h5py.Dataset = f[self._data_path]
-                return np.arange(data.shape[0]).tolist()
+        if self._image_key_path is None:
+            data: h5py.Dataset = self._h5file[self._data_path]
+            return np.arange(data.shape[0]).tolist()
```

Comment:
These lines are not covered by tests (I ran a coverage). I think it's an important special case that you should cover.

Reply:
Good catch, thanks! I do remember noticing this when I was making changes on the `previewing` branch after I made this PR.

The work on the `previewing` branch included moving the data indices calculation out of `StandardTomoLoader` into a `Preview` object. At that time, I then added a test to the `Preview` object to handle the case when the image key path is `None`, see the `test_preview_with_no_image_key()` test in `test_loader.py` in the PR for adding previewing https://github.com/DiamondLightSource/httomo/pull/209/files#diff-93229d65c37eac8fb2c218f0a1f0ebeb3a2fadf848ffd6557bf1375e7942dac3R769. (Apologies, the link won't automatically jump to the highlighted function due to the file diff being too large to show by default, one needs to manually click to show the diff for the `test_loader.py` file for it to then jump to the highlighted test function).

So I'm going to leave out a test for the loader to handle that case in this PR, because the PR for previewing will tackle it, hope that makes sense.

Reply:
sounds good